library/global.ml

```diff
@@ -59,12 +59,12 @@ let globalize f =
 
 let globalize0_with_summary fs f =
   let env = f (safe_env ()) in
-  Summary.unfreeze_summaries fs;
+  Summary.unfreeze_summaries ~partial:true fs;
```

Comment:
I tried to understand what this undocumented flag does, I even looked back at the PR which introduced it, but I don't get it. Here you use it, so maybe you understood it. Why is it needed and what does it do?

Reply:
After some digging, here there is a need and a lurking bug unrelated to this PR:
- the need of restoring only a subset of the system state (summary), which seems legit
- a hack to reset/undo some system component even if it is not part of the installed state. This need/workaround dates 2000 by JCF, so I'm clueless what that is about. Clearly, if you restore a partial state you don't want to touch the other parts.